DeviceBridgeE2E/EchoAzureFunction/DeviceBridgeE2EEcho/DeviceBridgeE2EEcho.cs

```diff
@@ -51,7 +51,7 @@ public static class DeviceBridgeE2EEcho
             } else if(req.Method == "DELETE")
             {
                 string value;
-                if (!cache.TryRemove(deviceId, out value))
+                if (cache.ContainsKey(deviceId) && !cache.TryRemove(deviceId, out value))
```

Comment:
The body of this if is never gonna execute. `TryRemove` only returns false if the key didn't exist (also it's a concurrent dictionary, so testing and removing in different commands is not atomic).
At the end, looks like you can replace the entire `DELETE` path with just `cache.TryRemove(deviceId, out string value)`.

Reply:
Let me know if you wanna have a quick sync on this one.